draft-kasselman-oauth_spiffe.md

```diff
@@ -223,8 +235,22 @@ An authorization server MAY register a client with a default set of scopes or re
 ## Grant Types
 Authorization servers MAY choose to limit the grant types for which the "register on first use" pattern is supported. This may be recorded as the "grant_type" metadata field.
 
-# Client Lifecycle Management
-The number of clients registered with an authorization server may grow significantly over time. An authorization MAY unregistered a client if it has not been used for an extended period of time to reduce the size of the client registration database.
+# Post-Registration Client Lifecycle Management
+After registration, there MUST be an initial client record with a direct link between the SPIFFE identifier in the SPIFFE credentials and the OAuth client identifier. However, additional work MAY be required to make the client operational, such as missing configuration or entitlement. This is particularly relevant for complex enterprise environments.
+
+## Configuration
+After registration, a client must be configured with the necessary operational metadata to function correctly and securely. An authorization server may use a number of mechanisms for obtaining metadata. Metadata may be statically preconfigured, automatically or manually created, or retrieved from a configuration management system. This can happen instantaneously after registration or it can be asynchronous.
```

Comment:
A client MUST be configured with the necessary operational metadata to function correctly and securely. An authorization server may use a number of mechanisms for obtaining metadata. Metadata may be statically preconfigured, automatically or manually created, or retrieved from a configuration management system. This can happen instantaneously after registration or it can be asynchronous. If metadata is added asynchronously, the authorization server MUST return an "Incomplete Registration" error whenever the client interacts with the authorization server, until the additional metadata has been added.